src/actions/receive-profile.js

```diff
@@ -1015,73 +993,56 @@ export async function checkIfWebChannelUsableForSymbolication(): Promise<boolean
 export function retrieveProfileFromBrowser(): ThunkAction<Promise<void>> {
   return async (dispatch) => {
     try {
-      let shouldUseWebChannel = false;
-      try {
-        shouldUseWebChannel = await Promise.race([
-          querySupportsGetProfileAndSymbolicationViaWebChannel(),
-          makeTimeoutRejectionPromise(5000),
-        ]);
-      } catch (e) {
-        if (e instanceof TimeoutError) {
-          // The browser never reacted to our WebChannelMessageToChrome event.
-          // This can happen if we're running on a browser that's not Firefox, or if we're running
-          // on an old version of Firefox which does not have support for any WebChannels.
-          // Ignore the timeout and fall back to window.geckoProfilerPromise below.
-        } else {
-          // The WebChannel responded with an error. This usually means that this profiler
-          // instance is running on a different host than the one that's specified in the
-          // preference `devtools.performance.recording.ui-base-url`.
-          // That's unusual, because we get here only for the /from-browser entry point,
-          // which is usually accessed from the new tab that the browser opens when a
-          // profile is captured - and that tab gets opened on the profiler instance that's
-          // configured in the base URL.
-          // We completely stop trying to connect to the browser and display an error message.
-          // (We don't fall back to the frame script, because we'd most likely display
-          // the waiting animation indefinitely and never connect to a frame script.)
-          throw new Error(oneLine`
-            This profiler instance was unable to connect to the
-            WebChannel. This usually means that it’s running on a
-            different host from the one that is specified in the
-            preference devtools.performance.recording.ui-base-url. If
-            you would like to capture new profiles with this instance, you can go to about:config
-            and change the preference. Error: ${e.name}: ${e.message}
-          `);
-        }
+      // Attempt to establish a connection to the browser.
+      // Disable the userAgent check by supplying a fake userAgent that
+      // pretends we're Firefox. This will make us attempt to establish
+      // a connection to the WebChannel even if we're running in the test
+      // suite.
+      const connectionStatus = await createBrowserConnection('Firefox/123.0');
+      switch (connectionStatus.status) {
+        case 'ESTABLISHED':
+          // Good. This is the normal case.
+          break;
+        // The other cases are error cases.
+        case 'NOT_FIREFOX':
+          throw new Error('/from-browser only works in Firefox browsers');
+        case 'WAITING':
+          throw new Error('unexpected');
```

Comment:
nit: can you please write a longer string, at least mentioning the value "WAITING" ?

Reply:
Changed to `'unexpected WAITING from createBrowserConnection'`.